josh-proxy/src/bin/josh-proxy.rs

```diff
@@ -81,6 +82,28 @@ impl JoshProxyUpstream {
     }
 }
 
+enum GitEnvVar {
```

Comment:
I don't see the advantage of introducing this type over just using the string constants in place. What's the reasoning?

Reply:
I'd prefer just keeping the strings in place like they were. Typos don't seem like a real threat to me. And having an extra thing to look up to see what something is just hurts readability. Even more so with the extra complication of going through a trait for conversion.
Also now there is a mix of some strings being inline constants and some being in this enum. Even just among the git env vars. Strings are simple. Let's just keep it simple.